client/src/components/Navbar.js

```diff
@@ -111,50 +112,63 @@ const Navbar = () => {
     }
 
     // checking for code after linkedin login
+    const [authCode, setAuthCode] = useState(null)
+
     useEffect(() => {
         async function checkForLinkedinToken() {
             const windowUrl = window.location.href
             if (windowUrl.includes('code=')) {
                 const codeMatch = windowUrl.match(/code=([a-zA-Z0-9_-]+)/)
+                if (!codeMatch) return
 
-                try {
-                    // Assuming codeMatch[1] contains the auth_code you wish to include in the request
-                    const authCode = codeMatch[1] // Ensure this variable is correctly defined in your context
-
-                    // Prepare the URL and headers for the fetchWithAuth call
-                    const url = `${HOST}/api/user/linkedin/userinfo`
-                    const headers = {
-                        auth_code: authCode, // Since fetchWithAuth handles the Authorization header, we only need to include additional headers
-                    }
-
-                    // Call fetchWithAuth to perform the request
-                    const data = await fetchWithAuth({
-                        url,
-                        method: 'GET',
-                        headers,
-                    })
-
-                    // Use the response data as needed
-                    setUserLinkedinInfo(data)
-                } catch (error) {
-                    console.error(error.message)
-                }
+                setAuthCode(codeMatch[1])
             }
         }
 
         checkForLinkedinToken()
     }, [])
 
+    useEffect(() => {
+        async function handleLinkedinToken() {
+            if (!authCode) return
+
+            try {
+                const url = `${HOST}/api/user/linkedin/userinfo`
+                const headers = { auth_code: authCode }
+
+                const { token, ...data } = await fetchWithAuth({
+                    url,
+                    method: 'GET',
+                    headers,
+                })
+
+                setUserLinkedinInfo(data)
+                localStorage.setItem('linkedinToken', token)
+                generateToken()
+
+                if (user) {
+                    updateProfile(user.profileId)
+                } else {
+                    login(user.email)
+                }
+            } catch (error) {
+                console.error(error.message)
+            }
+        }
+
+        handleLinkedinToken()
+    }, [authCode]) // Run when authCode or user changes
+
```

Comment:
clean. 🔥

Reply:
This was the useEffect that had me up till 1:30 AM 😭